tools/cmake/toolchains/ARM.cmake

```diff
@@ -8,49 +8,39 @@ set(CMAKE_AR "armar")
 set(ARM_ELF2BIN "fromelf")
 set_property(GLOBAL PROPERTY ELF2BIN ${ARM_ELF2BIN})
 
-# Sets toolchain options
-function(mbed_set_toolchain_options target)
-    list(APPEND common_options
-        "-c"
-        "--target=arm-arm-none-eabi"
-        "-mthumb"
-        "-Wno-armcc-pragma-push-pop"
-        "-Wno-armcc-pragma-anon-unions"
-        "-Wno-reserved-user-defined-literal"
-        "-Wno-deprecated-register"
-        "-fdata-sections"
-        "-fno-exceptions"
-        "-fshort-enums"
-        "-fshort-wchar"
-    )
+# tell cmake about compiler targets.
+# This will cause it to add the --target flag.
+set(CMAKE_C_COMPILER_TARGET arm-arm-none-eabi)
+set(CMAKE_CXX_COMPILER_TARGET arm-arm-none-eabi)
 
-    target_compile_options(${target}
-        INTERFACE
-            $<$<COMPILE_LANGUAGE:C>:${common_options}>
-    )
-
-    target_compile_options(${target}
-        INTERFACE
-            $<$<COMPILE_LANGUAGE:CXX>:${common_options}>
-    )
+# Sets toolchain options
+list(APPEND common_options
+    "-mthumb"
+    "-Wno-armcc-pragma-push-pop"
+    "-Wno-armcc-pragma-anon-unions"
+    "-Wno-reserved-user-defined-literal"
+    "-Wno-deprecated-register"
+    "-fdata-sections"
+    "-fno-exceptions"
+    "-fshort-enums"
+    "-fshort-wchar"
+)
 
-    target_compile_options(${target}
-        INTERFACE
-            $<$<COMPILE_LANGUAGE:ASM>:--target=arm-arm-none-eabi -masm=auto>
-    )
+list(APPEND asm_compile_options
+    -masm=auto
+    --target=arm-arm-none-eabi
```

Comment:
there's not CMAKE_ASM_COMPILER_TARGET as we do above for C/CXX ?

Reply:
I don't believe so... though it kinda depends on how ARM implemented the CMake scripts for ARM Compiler.  You could try adding it (and removing the manual --target flag) and see if it works.